### PR TITLE
WT-2093 Use the C99 bool type to clarify when functions return true/false

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

**Comment by michaelcahill** [comment]
@keithbostic, you are most likely to have a strong reaction to this change -- what do you think?

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by keithbostic** [comment]
I have no objection to this change, it makes sense to me.

I think we should probably agree to including any C99 features, rather than discussing them one-by-one?

**Comment by agorrod** [comment]
> I think we should probably agree to including any C99 features

I've opened https://jira.mongodb.org/browse/WT-2095 to cover a discussion about that.

**Comment by michaelcahill** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
@agorrod, can you please review?

File: src/cache/cache_las.c [hunk 1]
```diff
@@ -169,7 +169,7 @@ __wt_las_set_written(WT_SESSION_IMPL *session)
  * __wt_las_is_written --
  *	Return if the lookaside table has been written.
  */
-int
+bool
 __wt_las_is_written(WT_SESSION_IMPL *session)
 {
 	return (S2C(session)->las_written);
```
Comment (agorrod) [inline src/cache/cache_las.c]: Should ```las_written``` switch to a bool?

**Comment by agorrod** [comment]
@michaelcahill I had one nit, otherwise lgtm.

**Comment by michaelcahill** [comment]
Seems like we're all in agreement, let's merge to develop and see where we end up.

Note that the two bugs identified in the ticket are not fixed by this commit: there is more work to do before it can be resolved.

**Pull request merged**